src/noise/session.rs

```diff
@@ -223,14 +225,17 @@ impl Session {
             let mut nonce = [0u8; 12];
             nonce[4..12].copy_from_slice(&sending_key_counter.to_le_bytes());
             data[..src.len()].copy_from_slice(src);
-            seal_in_place(
-                &self.sender,
-                Nonce::assume_unique_for_key(nonce),
-                Aad::from(&[]),
-                &mut data[..src.len() + AEAD_SIZE],
-                AEAD_SIZE,
-            )
-            .unwrap()
+            self.sender
+                .seal_in_place_separate_tag(
```

Comment:
I think using seal_in_place_append_tag is the more appropriate function here?

Reply:
We can't use it, the destination buffer is a slice so doesn't implement Extend.

Reply:
OK.